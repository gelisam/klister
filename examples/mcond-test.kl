#lang "n-ary-app.kl"

(import (shift "mcond.kl" 1))

(define-macros
  ([test
    (lambda (stx)
      (syntax-case stx
        [(list (_ id))
         (mcond
          [(free-identifier=? id 'if) (pure '1)]
          [(free-identifier=? id 'else) (pure '2)]
          [(free-identifier=? id 'lambda) (pure '3)]
          [(pure #t) (pure '4)])]))]))

(example (test if))
(example (test else))
(example (test lambda))
(example (test define))
(example (test argh))
