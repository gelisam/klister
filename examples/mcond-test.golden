Signal 1 : Signal
Signal 2 : Signal
Signal 3 : Signal
Signal 4 : Signal
Signal 4 : Signal
